src/apps/search-header/search-header.tsx

```diff
@@ -1,27 +1,156 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
+import { useCombobox, UseComboboxStateChange } from "downshift";
+import {
+  SuggestionsFromQueryStringQuery,
+  useSuggestionsFromQueryStringQuery
+} from "../../core/dbc-gateway/generated/graphql";
 import SearchBar from "../../components/search-bar/search-bar";
+import { Autosuggest } from "../../components/autosuggest/autosuggest";
+import { Suggestion } from "../../components/autosuggest-text/autossugest-text";
 
 export interface SearchHeaderProps {
   searchHeaderUrl?: string;
   altText?: string;
-  inputPlaceholder?: string;
+  inputPlaceholderText?: string;
+  stringSuggestionAuthorText?: string;
+  stringSuggestionWorkText?: string;
+  stringSuggestionTopicText?: string;
 }
 
 const SearchHeader: React.FC<SearchHeaderProps> = ({
   searchHeaderUrl = "/search",
   altText = "search icon",
-  inputPlaceholder = "Search here"
+  inputPlaceholderText = "Search here",
+  stringSuggestionAuthorText = "Author",
+  stringSuggestionWorkText = "Work",
+  stringSuggestionTopicText = "Topic"
 }) => {
-  const [q, setQ] = useState("");
+  const [q, setQ] = useState<string>("");
+  const [suggestItems, setsuggestItems] = useState<any[]>([]);
+  const [currentlySelectedItem, setCurrentlySelectedItem] = useState<any>("");
+  const [isAutosuggestOpen, setIsAutosuggestOpen] = useState<boolean>(false);
+  const {
+    data,
+    isLoading,
+    status
+  }: {
+    data: SuggestionsFromQueryStringQuery | undefined;
+    isLoading: boolean;
+    status: string;
+  } = useSuggestionsFromQueryStringQuery({ q });
+
+  // once the query returns data, we set it into our useSate
+  useEffect(() => {
+    if (data) {
+      const arayOfResults = data.suggest.result;
+      setsuggestItems(arayOfResults);
+    }
+  }, [data]);
+
+  // if there are at least 3 chars in the search-field, we open the autosuggest
+  useEffect(() => {
+    if (q) {
+      if (q.length > 2) {
+        setIsAutosuggestOpen(true);
+      }
+    } else {
+      setIsAutosuggestOpen(false);
+    }
+  }, [q]);
```

Comment:
I do not understand this logic. This toggles the visibility of the autosuggest based on the length. I would expect the visibility should be based on whether we have any results back.

Reply:
This was a part of MK2 
![image](https://user-images.githubusercontent.com/28546954/174498722-f6925be9-2ade-4f43-a74a-fee90a37aa12.png)

The dropdown is open after 3 characters have been typed in even if we don't have the data yet. Then we just show a brief loading text.